content/hardware/05.pro-solutions/solutions-and-kits/edge-control/tutorials/smart-irrigation-system/content.md

```diff
@@ -0,0 +1,493 @@
+
+```arduino
+  water_level = vals->water_level_local;
+  water_volume = PI * (radius) * (radius) * (int)water_level * 10;
+
+  if (waterCtrl == 0) {
+    currentWater = water_volume;
+    Serial.println("Initial Water Volumen: ");
+    Serial.println(currentWater);
+    water_usage = 0;
+    waterCtrl = 1;
+  }
+
+  if (water_volume < currentWater) {
+    water_usage += currentWater - water_volume;
+    currentWater = water_volume;
+  } else if (water_volume > currentWater) {
+    currentWater = water_volume;
+  }
+```
+
+### The Cloud Dashboard
+
+Taking advantage of the Arduino Cloud, we can seamlessly integrate a simple but powerful dashboard to monitor and visualize the status of the system in real-time, resulting in a professional Human-Computer Interaction (HCI) as can be seen below:
+
+![Arduino Cloud project dashboard ](assets/dashboard.png)
+
+Within the Arduino Cloud's dashboard, the system variables can be monitored and controlled. We have temperature, humidity and rain probability gauges showing the current weather status and forecast, toggle switches to control each motorized valve accompanied by a scheduler widget to set automatic irrigation routines, also we have time series graphs showing the daily activated time of each valve. Finally, there are water supply monitoring widgets, one that shows the current water level in percentage from 0-100%, a water volume widget that shows the raiming liters of water in the tank, and a water usage widget that shows the liters used daily.
+
+![Arduino Cloud project dashboard on a smartphone](assets/dashboard-cel.png)
+
+We can easily access this dashboard from a PC, mobile phone or tablet from anywhere, receiving an instantaneous update wherever we are.
+
+## Full Smart Irrigation System Example
+
+All the necessary files to replicate this application note can be found below:
+
+- The complete code can be downloaded [here](assets/Edge-Control_MKR_Codes.zip)
+
+### Project Working Images
+
+![Zone 1 manual irrigation](assets/manual-irrigation.png)
+![Automatic irrigation timer countdown in zone 2 (1.4 minutes left)](assets/timer-LCD.png)
+![Zone 2 after the motorized valve is opened](assets/zone2.png)
+![Zone 2 automatic irrigation](assets/irrigation-process.jpg)
+![Irrigation in all zones](assets/4zones.gif)
+
+## Conclusion
+
+In this application note, we have learned how to build a smart irrigation system to water our crops automatically, manually and remotely, also analyzing the weather to avoid irrigation when it's raining so we can save water, using an Arduino Edge Control allows us to easily implement this kind of agriculture systems ready for field deployment. It is known that normally in remote areas we don't have WiFi access but this is to showcase the Edge Control capabilities to connect to the Cloud and be remotely controlled. The Arduino Pro product series is a great choice to develop robust and industrial environment proof solutions.
```

Comment:
```suggestion
In this application note, we have learned how to build a smart irrigation system to water our crops automatically, manually and remotely, also analyzing the weather to avoid irrigation when it's raining so we can save water, using an Arduino Edge Control allows us to easily implement this kind of agriculture systems ready for field deployment. It is known that normally in remote areas we don't have Wi-Fi® access but this is to showcase the Edge Control capabilities to connect to the cloud and be remotely controlled. The Arduino Pro product series is a great choice for developing robust and industrial environment-proof solutions.
```